### DISCUSSION TITLE
User Account for Self-Hosted Runner on Linux

### DISCUSSION THREAD

**Discussion by ghost** [community/community#24643] [Apps, API and Webhooks]
<p>I’m installing build runners on Linux VMs in our org, and configuring them to run as a service. The instructions are mostly clear, but there are three things that are confusing me:</p>
<p>First question:<br>
I’m installing the runner to /actions-runner. I can download &amp; unpack it just fine, so long as I’m using sudo. But when I go to configure the runner using sudo, it tells me “Must not run as sudo.” But if I run it without sudo, I don’t have access:</p>
<p>touch: cannot touch ‘.env’: Permission denied<br>
./env.sh: line 37: .path: Permission denied<br>
./env.sh: line 32: .env: Permission denied<br>
Unhandled exception. System.UnauthorizedAccessException: Access to the path ‘/actions-runner/_diag’ is denied.</p>
<p>I see some people online suggest setting RUNNER_ALLOW_RUNASROOT to get past this; but I feel like I must be doing something wrong. Am I installing the runner in the wrong place? Should I or should I not be using sudo?</p>
<p>Second question:<br>
If I bypass the above issue and configure the runner using sudo, then when I go to install it as a service it says it’s going to run using my personal Linux user account. Shouldn’t it create a new “github_runner” system account or something–similar to how the Windows build runner uses the NetworkService account? Do I need to create that user account manually? Are there instructions, guidance, or best practices around which account to use and how to do it? I can’t seem to find anything anywhere …</p>
<p>Third question:<br>
Even if I just allow it to run as my user account, and after calling “./svc.sh start”, the Linux runners still appear as “Offline” in GitHub. What else am I missing?</p>

**Comment by ghost** [comment, marked as answer]
<p>So after some experimentation, I got it working. Here’s what I’ve found:</p>
<p>The proper approach seems to be to create a “github_runner” user account first, then sudo into that user account and install the runner to ~/action-runner. My three issues all seem to have stemmed from the fact I was trying to install off the root (/) instead of home (~).</p>
<p>So I guess the main takeaway from this post is that I think there should be some additional documentation when setting up a Linux build runner specifically (<a href="https://github.com/organizations/" rel="noopener nofollow ugc">https://github.com/organizations/</a>&lt;org&gt;/settings/actions/runners/new?arch=x64&amp;os=linux), letting users know to create a user account first, sudo into it, and install the runner to the user account’s home (~). This differs from the Windows install directions–which recommend installing to C:\action-runner–and I think is worth calling out specifically to avoid confusion.</p>

**Reply by newfunda** [reply]
@rhollberg_r1github 
After runner is re-started, we are back to the same issue. I see below user is used even after following your step, but it worked first time when runner is already started and this works with any user, I tried with ec2-user and that also worked. But after re-start it fails.

User:
- When ec2 is already running: github_runner
- After ec2 restarts: /home/DevAISUser/actions-runner

Log below:

Getting Git version info
  Working directory is '/home/github_runner/actions-runner/_work/repo/repo'
  /usr/bin/git version
  git version 2.40.1
Temporarily overriding HOME='/home/github_runner/actions-runner/_work/_temp/7b2be55a-b0cf-46[20](https://github.com/org/repo/actions/runs/6804240384/job/18501334111#step:2:21)-b331-05157a800b8b' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
/usr/bin/git config --global --add safe.directory /home/github_runner/actions-runner/_work/repo/repo
